packages/visx-xychart/src/hooks/useDataRegistry.ts

```diff
@@ -1,12 +1,42 @@
 import { AxisScale } from '@visx/axis';
-import { useMemo } from 'react';
+import { useCallback, useMemo, useState } from 'react';
 import DataRegistry from '../classes/DataRegistry';
 
 /** Hook that returns a constant instance of a DataRegistry.  */
 export default function useDataRegistry<
   XScale extends AxisScale,
   YScale extends AxisScale,
-  Datum = unknown
+  Datum extends object
 >() {
-  return useMemo(() => new DataRegistry<XScale, YScale, Datum>(), []);
+  const [, forceUpdate] = useState(Math.random());
+  const privateRegistry = useMemo(() => new DataRegistry<XScale, YScale, Datum>(), []);
+
+  const registerData = useCallback(
```

Comment:
as all of these 5 functions depends only on `privateRegistry`, perhaps a single `useMemo` block is enough?
or should the first two also depend on the value of `forceUpdate`?

Reply:
good call! initially I didn't have the force update, but you're right that now a single memo works 👌 had to wrap some of the functions for the appropriate `this` context to work.